third-party/imgui/realsense_imgui.h

```diff
@@ -0,0 +1,44 @@
+/* License: Apache 2.0. See LICENSE file in root directory. */
+/* Copyright(c) 2024 Intel Corporation. All Rights Reserved. */
+#pragma once
+
+#include "imgui.h"
+#include "imgui_internal.h"
+
+namespace RsImGui
+{
+    class ScopePushFont
+    {
+    public:
+        ScopePushFont(ImFont* font) { ImGui::PushFont(font); }
+        ~ScopePushFont() { ImGui::PopFont(); }
+    };
+    class ScopePushStyleColor
+    {
+    public:
+        ScopePushStyleColor(ImGuiCol idx, const ImVec4& col) { ImGui::PushStyleColor(idx, col); }
+        ~ScopePushStyleColor() { ImGui::PopStyleColor(); }
+    };
+    class ScopePushStyleVar
+    {
+    public:
+        ScopePushStyleVar(ImGuiStyleVar idx, float val) { ImGui::PushStyleVar(idx, val); }
+        ScopePushStyleVar(ImGuiStyleVar idx, const ImVec2& val) { ImGui::PushStyleVar(idx, val); }
+        ~ScopePushStyleVar() { ImGui::PopStyleVar(); }
+    };
+    bool          SliderIntWithSteps(const char* label, int* v, int v_min, int v_max, int v_step = 1);
+    float         RoundScalar(float value, int decimal_precision);
+    bool          CustomComboBox(const char* label, int* current_item, const char* const items[], int items_count);
+    bool          SliderBehavior(const ImRect& frame_bb, ImGuiID id, float* v, float v_min, float v_max, float power, int decimal_precision, ImGuiSliderFlags flags, bool render_bg);
+    bool          VSliderFloat(const char* label, const ImVec2& size, float* v, float v_min, float v_max, const char* display_format, float power, bool render_bg);
+    bool          SliderIntTofloat(const char* label, int* v, int v_min, int v_max, const char* display_format = "%.0f");
+    void          PushNewFrame();
+    void          PopNewFrame();
+
+}
+
+#define CONCAT_(x,y) x##y
+#define CONCAT(x,y) CONCAT_(x,y)
+#define ImGui_ScopePushFont(f) RsImGui::ScopePushFont CONCAT(scope_push_font, __LINE__) (f)
+#define ImGui_ScopePushStyleColor(idx, col) RsImGui::ScopePushStyleColor CONCAT(scope_push_style_color, __LINE__) (idx, col)
```

Comment:
Override?

Reply:
Not an override added comment to explain the macros functionality